docs/how_to/aggregator.md

```diff
@@ -0,0 +1,159 @@
+
+### Coordinator configuration
+
+Metrics will still arrive at the m3coordinator, they simply need to be forwarded to m3aggregators. The m3coordinator then also needs to receive metrics that have been aggregated from the m3aggregator and store them in M3DB, so running an ingestion server should be configured.
+
+Here is the config you should add to your m3coordinator:
+```
+# This is for sending metrics to the remote m3aggregators
+downsample:
+  remoteAggregator:
+    client:
+      placementKV:
+        namespace: /placement
+      placementWatcher:
+        key: m3aggregator
+        initWatchTimeout: 10s
+      hashType: murmur32
+      shardCutoffLingerDuration: 1m
+      flushSize: 1440
+      maxTimerBatchSize: 1120
+      queueSize: 10000
+      queueDropType: oldest
+      encoder:
+        initBufferSize: 2048
+        maxMessageSize: 10485760
+        bytesPool:
+          buckets:
+            - capacity: 2048
+              count: 4096
+            - capacity: 4096
+              count: 4096
+          watermark:
+            low: 0.7
+            high: 1.0
+      connection:
+        writeTimeout: 250ms
+
+# This is for configuring the ingestion server that will receive metrics from the m3aggregators on port 7507
+ingest:
+  ingester:
+    workerPoolSize: 100
+    opPool:
+      size: 100
+    retry:
+      maxRetries: 3
+      jitter: true
+    logSampleRate: 0.01
+  m3msg:
+    server:
+      listenAddress: "0.0.0.0:7507"
+      retry:
+        maxBackoff: 10s
+        jitter: true
+```
+
+### Topology
+
+Once you have your m3aggregators running, you can setup a m3aggregator topology by issuing a request to your coordinator (be sure to use your own hostnames, number of shards and replication factor):
+```
+curl -vvvsSf -X POST http://m3coordinator01:7201/api/v1/services/m3aggregator/placement/init -d '{
+    "num_shards": 64,
+    "replication_factor": 2,
+    "instances": [
+        {
+            "id": "m3aggregator01:6000",
+            "isolation_group": "availability-zone-a",
+            "zone": "embedded",
+            "weight": 100,
+            "endpoint": "m3aggregator01:6000",
+            "hostname": "m3aggregator01",
+            "port": 6000
+        },
+        {
+            "id": "m3aggregator02:6000",
+            "isolation_group": "availability-zone-b",
+            "zone": "embedded",
+            "weight": 100,
+            "endpoint": "m3aggregator02:6000",
+            "hostname": "m3aggregator02",
+            "port": 6000
+        }
+    ]
+}'
+```
+
+### Aggregation topic and m3coordinator discovery
+
+#### Create aggregation topic
+
+Now we must setup a topic for the m3aggregator to send computed metrics aggregations back to an m3coordinator instance for storage to M3DB:
+```
+curl -vvvsSf -X POST http://m3coordinator01:7201/api/v1/topic/init -d '{
+    "numberOfShards": 64
+}'
+```
```

Comment:
Should the default topic name be mentioned? Looking through the source code, the default topic name used is `aggregated_metrics`. And there seems to be a way to specify the topic by setting `topic-name` in the http request header.

Reply:
Yes we should definitely mention that.

Reply:
Fixed.